From b78316cee968d2b448837cc391940e984cfff248 Mon Sep 17 00:00:00 2001
From: zhangmeng <zhangmeng.kevin@spacemit.com>
Date: Tue, 16 Apr 2024 16:29:07 +0800
Subject: [PATCH 0619/1448] clear compile warning:

drivers/net/wireless/realtek/rtl8852be/os_dep/osdep_service.c:247:13: warning: conflicting types for 'dbg_rtw_vmfree' due to enum/integer mismatch; have 'void(void *, u32,  const enum mstat_f,  const char *, const int)' {aka 'void(void *, unsigned int,  const enum mstat_f,  const char *, const int)'} [-Wenum-int-mismatch]
  247 | inline void dbg_rtw_vmfree(void *pbuf, u32 sz, const enum mstat_f flags, const char *func, const int line)
      |             ^~~~~~~~~~~~~~
In file included from /data/home2/zhangmeng/WorkSpace/k1x-dev/linux-6.1/drivers/net/wireless/realtek/rtl8852be/include/drv_types.h:25,
                 from drivers/net/wireless/realtek/rtl8852be/os_dep/osdep_service.c:17:
/data/home2/zhangmeng/WorkSpace/k1x-dev/linux-6.1/drivers/net/wireless/realtek/rtl8852be/include/osdep_service.h:145:6: note: previous declaration of 'dbg_rtw_vmfree' with type 'void(void *, const enum mstat_f,  u32,  const char *, const int)' {aka 'void(void *, const enum mstat_f,  unsigned int,  const char *, const int)'}
  145 | void dbg_rtw_vmfree(void *pbuf, const enum mstat_f flags, u32 sz, const char *func, const int line);
      |      ^~~~~~~~~~~~~~

Change-Id: Ia260d89ce4ee6bc3c2215811dc5c9b2af7eeb70e
---
 .../net/wireless/realtek/rtl8852be/include/osdep_service.h    | 4 ++--
 .../realtek/rtl8852be/phl/hal_g6/phy/bb/halbb_physts.h        | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/wireless/realtek/rtl8852be/include/osdep_service.h b/drivers/net/wireless/realtek/rtl8852be/include/osdep_service.h
index 29269524591b..ea7c02fe48b9 100755
--- a/drivers/net/wireless/realtek/rtl8852be/include/osdep_service.h
+++ b/drivers/net/wireless/realtek/rtl8852be/include/osdep_service.h
@@ -142,10 +142,10 @@ void rtw_mstat_dump(void *sel);
 bool match_mstat_sniff_rules(const enum mstat_f flags, const size_t size);
 void *dbg_rtw_vmalloc(u32 sz, const enum mstat_f flags, const char *func, const int line);
 void *dbg_rtw_zvmalloc(u32 sz, const enum mstat_f flags, const char *func, const int line);
-void dbg_rtw_vmfree(void *pbuf, const enum mstat_f flags, u32 sz, const char *func, const int line);
+void dbg_rtw_vmfree(void *pbuf, u32 sz, const enum mstat_f flags, const char *func, const int line);
 void *dbg_rtw_malloc(u32 sz, const enum mstat_f flags, const char *func, const int line);
 void *dbg_rtw_zmalloc(u32 sz, const enum mstat_f flags, const char *func, const int line);
-void dbg_rtw_mfree(void *pbuf, const enum mstat_f flags, u32 sz, const char *func, const int line);
+void dbg_rtw_mfree(void *pbuf, u32 sz, const enum mstat_f flags, const char *func, const int line);
 
 struct sk_buff *dbg_rtw_skb_alloc(unsigned int size, const enum mstat_f flags, const char *func, const int line);
 void dbg_rtw_skb_free(struct sk_buff *skb, const enum mstat_f flags, const char *func, const int line);
diff --git a/drivers/net/wireless/realtek/rtl8852be/phl/hal_g6/phy/bb/halbb_physts.h b/drivers/net/wireless/realtek/rtl8852be/phl/hal_g6/phy/bb/halbb_physts.h
index c86ffee6fbe5..7806988a06f9 100755
--- a/drivers/net/wireless/realtek/rtl8852be/phl/hal_g6/phy/bb/halbb_physts.h
+++ b/drivers/net/wireless/realtek/rtl8852be/phl/hal_g6/phy/bb/halbb_physts.h
@@ -766,8 +766,8 @@ void halbb_physts_rpt_gen(struct bb_info *bb, u32 physts_bitmap,
 			  struct physts_result *rpt,
 			  bool physts_rpt_valid, struct physts_rxd *desc,
 			  bool is_cck_rate, bool is_ie8_valid);
-void halbb_physts_ie_bitmap_set(struct bb_info *bb, u32 ie_page, u32 bitmap);
-u32 halbb_physts_ie_bitmap_get(struct bb_info *bb, u32 ie_page);
+void halbb_physts_ie_bitmap_set(struct bb_info *bb, enum bb_physts_bitmap_t ie_page, u32 bitmap);
+u32 halbb_physts_ie_bitmap_get(struct bb_info *bb, enum bb_physts_bitmap_t ie_page);
 void halbb_physts_ie_bitmap_en(struct bb_info *bb, enum bb_physts_bitmap_t type,
 			       enum bb_physts_ie_t ie, bool en);
 void halbb_phy_sts_manual_trig(struct bb_info *bb, enum bb_mode_type mode, u8 ss);
-- 
2.47.0

